services/src/pro/projects/hashmap_projectdb.rs

```diff
@@ -33,12 +34,17 @@ impl ProjectDb<UserSession> for ProHashMapProjectDb {
             offset,
             limit,
         } = options.user_input;
+
+        let all_projects = self.projects.read().await;
+
         #[allow(clippy::flat_map_option)]
         let mut projects = self
             .permissions
+            .read()
```

Comment:
Why do you have in this case two locks but for other in memory dbs only one lock for everything?

Reply:
it requires only one more lock but saves introducing a backend struct, so: convenience